Code/Editor/TrackView/StringKeyUIControls.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+
+#include "EditorDefs.h"
+
+#include "KeyUIControls.h"
+#include "TrackViewKeyPropertiesDlg.h"                          // for CTrackViewKeyUIControls
+
+#include <CryCommon/Maestro/Types/AnimValueType.h>              // for AnimValueType
+
+bool CStringKeyUIControls::OnKeySelectionChange(const CTrackViewKeyBundle& selectedKeys)
+{
+    if (!selectedKeys.AreAllKeysOfSameType())
+    {
+        return false;
+    }
+
+    if (selectedKeys.GetKeyCount() == 1)
+    {
+        const CTrackViewKeyHandle& keyHandle = selectedKeys.GetKey(0);
+
+        AnimValueType valueType = keyHandle.GetTrack()->GetValueType();
+        if (valueType == AnimValueType::String)
+        {
+            IStringKey stringKey;
+            keyHandle.GetKey(&stringKey);
+            mv_value = QString::fromStdString(stringKey.m_strValue.c_str());            
+            return true;        
+        }
+    }
+    return false;
+}
+
+// Called when UI variable changes.
+void CStringKeyUIControls::OnUIChange(IVariable* pVar, CTrackViewKeyBundle& selectedKeys)
+{
+    CTrackViewSequence* sequence = GetIEditor()->GetAnimation()->GetSequence();
+
+    if (!sequence || !selectedKeys.AreAllKeysOfSameType())
+    {
+        return;
+    }
+
+    for (unsigned int keyIndex = 0; keyIndex < selectedKeys.GetKeyCount(); ++keyIndex)
+    {
+        CTrackViewKeyHandle keyHandle = selectedKeys.GetKey(keyIndex);
+
+        AnimValueType valueType = keyHandle.GetTrack()->GetValueType();
+        if (valueType == AnimValueType::String)
```

Comment:
maybe early continue to prevent too much nesting + valueType can be const (I prefer to be explicit for largely scoped variables)